### PR TITLE
Send FCM notifications on configured statuses exceed & Refactor persistence

### PR DESCRIPTION
**Pull request by adnanjpg**
- added a top level persistence module
- added `notification_logs` table
- added a function that triggers an FCM notification when any of the configured statuses exceed

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -105,6 +106,14 @@ async fn main() {
         }
     };
 
+    match crate::persistence::init_db().await {
```
Comment (adnanjpg) [inline src/main.rs:109]: the monitor module used to initialize its own persistence, now the top level persistence module initializes its own submodules (such as the `notification_logs` submodule) and all other nested persistence modules (such as the `monitor/persistence` module)

File: src/monitor/persistence/hardware_cpu_info.rs [hunk 2]
```diff
@@ -2,12 +2,12 @@ use sqlx::SqliteConnection;
...
 use crate::monitor::models::get_hardware_info::HardwareCpuInfo;
 
-use super::{get_sql_connection, FetchId, SQLITE_DB_CONN_STR};
+use super::{get_default_sql_connection, FetchId};
 
 const HARDWARE_CPU_INFOS_TABLE_NAME: &str = "cpu_infos";
 
 pub(super) async fn insert_hardware_cpu_info(info: &HardwareCpuInfo) -> Result<(), sqlx::Error> {
-    let mut conn = get_sql_connection(SQLITE_DB_CONN_STR).await?;
+    let mut conn = get_default_sql_connection().await?;
```
Comment (adnanjpg) [inline src/monitor/persistence/hardware_cpu_info.rs:10]: felt no need to let each module get a connection string and passing it to the function, as far as the vision of the project goes the db connection config is going to be from a single place

File: src/persistence/notification_logs.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
+use log::debug;
+use sqlx::SqliteConnection;
+
+use super::get_default_sql_connection;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, sqlx::Type, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum NotificationType {
```
Comment (adnanjpg) [inline src/persistence/notification_logs.rs:8]: single point of all types of notifications. when we need a new type, adding a variant here would be enough

File: src/main.rs (hunk 1 above)
Comment (isaidsari) [reply src/main.rs:109]: understandable

File: src/monitor/persistence/hardware_cpu_info.rs (hunk 2 above)
Comment (isaidsari) [reply src/monitor/persistence/hardware_cpu_info.rs:10]: since we have single db file, yeah that was unnecessary.

**Review by isaidsari — COMMENTED** [review]
looks good to me

File: src/monitor/persistence/hardware_cpu_info.rs (hunk 2 above)
Comment (adnanjpg) [reply src/monitor/persistence/hardware_cpu_info.rs:10]: even in the future if we add a feature to connect to 3rd party db sources, having the connection string in a single place will still be enough

**Pull request merged**